File: packages/elixir-sdk/test/conformance/sdk_conformance_test.exs
defmodule TreeDxSdk.ConformanceTest do
  use ExUnit.Case, async: true

  defp scenarios do
    "../sdk-spec/conformance/scenarios"
    |> File.ls!()
    |> Enum.filter(&String.ends_with?(&1, ".yaml"))
    |> Enum.flat_map(fn file ->
      text = File.read!(Path.join("../sdk-spec/conformance/scenarios", file))
      ids = Regex.scan(~r/^  - id: (.+)$/m, text) |> Enum.map(fn [_, id] -> id end)
      caps = Regex.scan(~r/^    capabilityId: (.+)$/m, text) |> Enum.map(fn [_, id] -> id end)
      Enum.zip(ids, caps) |> Enum.map(fn {id, cap} -> %{"id" => id, "capabilityId" => cap} end)
    end)
  end

  test "scenario catalog loads and reports not configured" do
    client = TreeDxSdk.Client.new(base_url: "http://localhost:4000")
    adapter = TreeDxSdk.Conformance.Adapter.new(client)
    scenarios = scenarios()
    assert scenarios != []

    for scenario <- scenarios do
      assert scenario["id"] != ""
      assert scenario["capabilityId"] != ""

      assert %{status: :not_configured} =
               TreeDxSdk.Conformance.Adapter.run_scenario(adapter, scenario)
    end
  end
end

File: packages/ts-sdk/test/conformance/sdk-conformance.test.ts
import fs from 'node:fs';
import path from 'node:path';
import YAML from 'yaml';
import { describe, expect, it } from 'vitest';
import { TreeDxClient, TreeDxConformanceAdapter, type TreeDxConformanceScenario } from '../../src/treedx/index.js';
import { MockTransport } from '../adapters/mock.js';

function loadScenarios(): TreeDxConformanceScenario[] {
  const scenariosDir = path.resolve(import.meta.dirname, '../../../sdk-spec/conformance/scenarios');
  return fs.readdirSync(scenariosDir)
    .filter((fileName) => fileName.endsWith('.yaml'))
    .flatMap((fileName) => YAML.parse(fs.readFileSync(path.join(scenariosDir, fileName), 'utf8')).scenarios);
}

describe('TreeDxConformanceAdapter', () => {
  it('loads shared scenario records', () => {
    const scenarios = loadScenarios();
    expect(scenarios.length).toBeGreaterThan(0);
    expect(scenarios.every((scenario) => scenario.id && scenario.capabilityId)).toBe(true);
  });

  it('reports not_configured without a server', async () => {
    const client = new TreeDxClient({ baseUrl: 'http://treedx.test', transport: new MockTransport() });
    const adapter = new TreeDxConformanceAdapter({ client });
    const [scenario] = loadScenarios();
    await expect(adapter.runScenario(scenario)).resolves.toMatchObject({ scenarioId: scenario.id, status: 'not_configured' });
  });
});
